### PR TITLE
secscan: Cache clair vuln reports (PROJQUAY-6057)

### PR DESCRIPTION
**Pull request by syed**
uses modelcache to for caching clair security report responses

### CODE REVIEW

**Comment by hdonnay** [comment]
Closes #1039 

cc @crozzy :eyes:

**Comment by crozzy** [comment]
:+1: LGTM I think manifest digest is sufficient as a key

File: data/cache/cache_key.py [hunk 1]
```diff
@@ -73,3 +73,13 @@ def for_appr_show_package(namespace, package_name, release, media_type, cache_co
     return CacheKey(
```
Comment (kleesc) [inline data/cache/cache_key.py:73]: nit: The appr cache keys are no longer used. Can we remove them?

**Review by kleesc — DISMISSED** [review]

File: data/cache/cache_key.py (hunk 1 above)
Comment (syed) [reply data/cache/cache_key.py:73]: yeah I'll remove them. I also see there's a circular dep on import I'll fix that too

Comment (kleesc) [reply data/cache/cache_key.py:73]: You'll probably need to pass around the model_cache from the caller

Comment (syed) [reply data/cache/cache_key.py:73]: yeah too bad, also have to pass `cache_key` singleton as well

File: data/secscan_model/test/test_secscan_v4_model.py [hunk 2]
```diff
@@ -149,6 +154,53 @@ def test_load_security_information_success(initialized_db, set_secscan_config):
...
 
+def test_load_security_information_success_with_cache(initialized_db, set_secscan_config):
+    model_cache = InMemoryDataModelCache(TEST_CACHE_CONFIG)
+    model_cache.empty_for_testing()
+
+    repository_ref = registry_model.lookup_repository("devtable", "simple")
+    tag = registry_model.get_repo_tag(repository_ref, "latest")
+    manifest = registry_model.get_manifest_for_tag(tag)
+
+    sec_info_cache_key = cache_key.for_security_report(manifest.digest)
```
Comment (hdonnay) [inline data/secscan_model/test/test_secscan_v4_model.py:165]:
```suggestion
    sec_info_cache_key = cache_key.for_security_report(manifest.digest, model_cache.cache_config)
```

**Review by hdonnay — APPROVED** [review]
LGTM when CI is green

**Pull request merged**